src/Backend/Account.vala

```diff
@@ -20,9 +20,38 @@
 
 public class Mail.Backend.Account : GLib.Object {
     public Camel.Service service { get; construct; }
+
     public Account (Camel.Service service) {
         Object (service: service);
     }
+
+    construct {
+        unowned var network_manager = GLib.NetworkMonitor.get_default ();
+        network_manager.network_changed.connect (() => {
+            manage_connection.begin (network_manager.network_available);
+        });
+    }
+
+    public async void manage_connection (bool online) {
+        var offlinestore = (Camel.OfflineStore)service;
+
+        if (online) {
+            try {
+                yield offlinestore.set_online (true, GLib.Priority.DEFAULT, null);
+                yield offlinestore.synchronize (false, GLib.Priority.DEFAULT, null);
+            } catch (Error e) {
+                critical (e.message);
+            }
+            return;
+        }
+
+        try {
+            yield offlinestore.set_online (false, GLib.Priority.DEFAULT, null);
+        } catch (Error e) {
+            critical (e.message);
```

Comment:
Should this be a user facing error? If we're unable to come online that seems like something we should throw a dialog for maybe?

Reply:
Probably should yeah